pkg/api/file_repository.go

```diff
@@ -36,21 +41,40 @@ func NewFileSystemRepository(dir string) (*FileSystemRepository, error) {
 			}
 
 			definition := repo.parseDefinition(appConfigBody)
-			if err = repo.Add(definition); err != nil {
-				log.WithError(err).Error("Can't add the definition to the repository")
-				return nil, err
+			for _, v := range definition.defs {
+				if err = repo.Add(&v); err != nil {
+					log.WithError(err).Error("Can't add the definition" + v.Name + " to the repository")
+					return nil, err
+				}
 			}
 		}
 	}
 
 	return repo, nil
 }
 
-func (r *FileSystemRepository) parseDefinition(apiDef []byte) *Definition {
-	appConfig := NewDefinition()
-	if err := json.Unmarshal(apiDef, appConfig); err != nil {
+//Unmarshal a single Definition
```

Comment:
Missing space `// Unmarshal`

Reply:
TBH I think this method should be inside of `parseDefinition` 
```go
func (r *FileSystemRepository) parseDefinition(apiDef []byte) definitionList {
	appConfigs := definitionList{}

	// try unmarshalling when json is an unnamed Array of multiple definitions
	if err := json.Unmarshal(apiDef, &appConfigs); err != nil {
		// try unmarshalling when json is a single Definition
		appConfigs.defs = append(appConfigs.defs, *NewDefinition())
        if err := json.Unmarshal(apiDef, &appConfigs.defs[0]); err != nil {
		    log.WithError(err).Error("[RPC] --> Couldn't unmarshal api configuration")
	    }
	}

	return appConfigs
}
```

Reply:
Hi, @rafaeljesus . Which method should be inside of parseDefinition?

Reply:
@marcelnubi I was suggesting to get rid of `unmarshalDefinition` method and move it's contents to `parseDefinition` WDYT?

Reply:
I think its right move. Actually, I'm not sure  why I wrote it like that in the first place :P